package.json

```diff
@@ -73,7 +73,8 @@
     "webpack-stream": "^4.0.0"
   },
   "resolutions": {
-    "@lerna/**/@lerna/collect-updates": "https://github.com/babel/lerna.git#babel-collect-updates"
+    "@lerna/**/@lerna/collect-updates": "https://github.com/babel/lerna.git#babel-collect-updates",
+    "@lerna/cli/yargs": "14.2.0"
```

Comment:
Version `14.2.1` generates an error:

```
TypeError: Cannot delete property '--' of #<Object>
    at Object.Yargs.self._copyDoubleDash (/home/nicolo/Documenti/dev/babel/babel/node_modules/@lerna/cli/node_modules/yargs/yargs.js:1196:5)
    at Object.parseArgs [as _parseArgs] (/home/nicolo/Documenti/dev/babel/babel/node_modules/@lerna/cli/node_modules/yargs/yargs.js:1097:60)
    at Object.parse (/home/nicolo/Documenti/dev/babel/babel/node_modules/@lerna/cli/node_modules/yargs/yargs.js:578:25)
    at main (/home/nicolo/Documenti/dev/babel/babel/node_modules/lerna/index.js:44:6)
    at Object.<anonymous> (/home/nicolo/Documenti/dev/babel/babel/node_modules/lerna/cli.js:11:15)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
```

because of https://github.com/babel/babel/blob/6ba113197f34b6a7e6e24c339c2d8d72a1fd9435/Makefile#L240

cc @bcoe (author of the commit in v14.2.1 - yargs/yargs@e78e76e3ac0551d4f30c71a05ddb21582960fcef) is this a known issue fixed in v15?

Reply:
Could we work around it by
```
node $(npm bin)/lerna bootstrap -- --ignore-engines
```

Reply:
@nicolo-ribaudo something outside of yargs, in `lerna` is calling `Object.seal()` on the `argv` object, before the parse is finished (there's a two pass parse that temporarily stores positional arguments in `--` and then merges them into the final parse).

Reply:
Nope:
```
node node_modules/.bin/lerna bootstrap -- --ignore-engines
/home/nicolo/Documenti/dev/babel/babel/node_modules/@lerna/cli/node_modules/yargs/yargs.js:1184
      else throw err
           ^

TypeError: Cannot delete property '--' of #<Object>
    at Object.Yargs.self._copyDoubleDash (/home/nicolo/Documenti/dev/babel/babel/node_modules/@lerna/cli/node_modules/yargs/yargs.js:1196:5)
    at Object.parseArgs [as _parseArgs] (/home/nicolo/Documenti/dev/babel/babel/node_modules/@lerna/cli/node_modules/yargs/yargs.js:1097:60)
    at Object.parse (/home/nicolo/Documenti/dev/babel/babel/node_modules/@lerna/cli/node_modules/yargs/yargs.js:578:25)
    at main (/home/nicolo/Documenti/dev/babel/babel/node_modules/lerna/index.js:44:6)
    at Object.<anonymous> (/home/nicolo/Documenti/dev/babel/babel/node_modules/lerna/cli.js:11:15)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
make: *** [Makefile:240: lerna-bootstrap] Error 1

```

Reply:
@nicolo-ribaudo could you try commenting out this line?

https://github.com/lerna/lerna/blame/2c487fe5f612739970eb08026b2bf7da82a32b66/core/command/index.js#L99

I haven't quite figured out how to make a reproduction in yargs; If we can figure out a reproduction, I'd be happy to add a `try`/`catch`, and just skip the `delete` step for this edge case.

This does not fail, like I would hope:

```
  // see: https://github.com/babel/babel/pull/10733
  it('does not fail if Object.seal() has been called on argv', () => {
    Object.seal(process.argv)
    const argv = yargs()
      .command('cmd <version>', 'a command', () => {}, (argv) => {
        Object.freeze(argv)
      }).argv
    console.info(argv);
  })
```

Reply:
Yup, commenting the `freeze` call makes it work.

Reply:
@bcoe I managed to reproduce it with this example:

```js
const argv = yargs()
    .command('cmd <version>', 'a command', () => {}, (argv) => {
        argv["--"] = ["x"]
        Object.freeze(argv)
    }).parse(["cmd", "1", "--", "foo"])
```

Manually mutating `ærgv` feels a bit hacky; I'm trying to understand what lerna does.

Reply:
.

Reply:
@nicolo-ribaudo thanks for the minimal reproduction, I'm convinced I would like to find a way to get rid of the delete hack, not quite sure what the best refactor is yet since the command logic is complex -- that whole part of the codebase might be due for an overall.

I am going to float this patch to 14.2.2:

https://github.com/yargs/yargs/pull/1483

Reply:
please try `+ yargs@14.2.2` -- it's not a perfect fix, but should unblock these issues with lerna.

Reply:
Thanks!

Reply:
Ack, sorry folks. I don't even remember why I cared about freezing the argv. I can remove it, as it serves basically no purpose.

Reply:
@evocateur I think you should be able to freeze the object once it's been passed to a command ... so it feels like yargs stepping on your toes as well.

I was thinking this morning, perhaps we pass a shallow clone of the object to the command rather than the actual `argv`.

Reply:
I'm fixing it in https://github.com/lerna/lerna/commit/8ca85a4f07acbec02d41077faacdd1f4a62e86a3 by deep cloning `argv` before I mutate it. Releasing momentarily...

Reply:
lerna [v3.18.5](https://github.com/lerna/lerna/releases/tag/v3.18.5) published with the aforementioned fix

Reply:
thanks @evocateur!